runtime/tests/jni/booleantest.c

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2018 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+#include "jnitest_internal.h"
+
+static jclass runtimeException = NULL;
+
+jboolean JNICALL
+Java_j9vm_test_booleans_BooleanTest_setUp(JNIEnv *env, jclass instanceClass)
+{
+	runtimeException = (*env)->FindClass(env, "java/lang/Runtime");
+
+	return (jboolean) (runtimeException != NULL);
+}
+
+void JNICALL
+Java_j9vm_test_booleans_BooleanTest_tearDown(JNIEnv *env, jclass instanceClass)
+{
+	(*env)->DeleteGlobalRef(env, runtimeException);
+}
+
+
+void JNICALL
+Java_j9vm_test_booleans_BooleanTest_putInstanceBoolean(JNIEnv *env, jclass clazz, jclass instanceClass, jobject instance, jstring fieldName, jstring fieldSig, jint value)
+{
+	jboolean isCopy = JNI_FALSE;
+	const char* fieldNameUTF = (*env)->GetStringUTFChars(env, fieldName, &isCopy);
+	const char* fieldSigUTF = (*env)->GetStringUTFChars(env, fieldSig, &isCopy);
+
+	jfieldID field = (*env)->GetFieldID(env, instanceClass, fieldNameUTF, fieldSigUTF);
```

Comment:
I strongly prefer to see this kind of code written in Java using standard reflection to lookup the field.  The JNI native  can use `jfieldID FromReflectedField(JNIEnv *env, jobject field)` to convert the reflect field to a jfield and avoid all the error handling.